sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -844,25 +949,45 @@ static int sof_card_dai_links_create(struct device *dev,
 		dmic_num, hdmi_num);
 
 	num_links = ssp_num + sdw_num + dmic_num + hdmi_num;
-	links = devm_kcalloc(dev, num_links, sizeof(struct snd_soc_dai_link),
-			     GFP_KERNEL);
-	cpus = devm_kcalloc(dev, num_links,
-			    sizeof(struct snd_soc_dai_link_component),
-			    GFP_KERNEL);
+	links = devm_kcalloc(dev, num_links, sizeof(*links), GFP_KERNEL);
+	i = ssp_num + cpu_dai_num + dmic_num + hdmi_num;
+	cpus = devm_kcalloc(dev, i, sizeof(*cpus), GFP_KERNEL);
```

Comment:
...continuing the above comment. At least store size of allocated array in "num_cpus" and pass that (and num_links) to the subfunctions that handle the array. Now they are going through the arrays blind, which seems pretty dangerous.

Reply:
thanks, defensive programming. I will refine it.

Reply:
ok, it wasn't only my stomach that didn't like this